pkg/registry/core/node/storage/storage_test.go

```diff
@@ -54,37 +54,59 @@ func newStorage(t *testing.T) (*REST, *etcd3testing.EtcdTestServer) {
 	return storage.Node, server
 }
 
-func validNewNode() *api.Node {
-	return &api.Node{
+type tweak func(*api.Node)
+
+func newNode(name string, tweaks ...tweak) *api.Node {
+	node := &api.Node{
 		ObjectMeta: metav1.ObjectMeta{
-			Name: "foo",
+			Name: name,
 			Labels: map[string]string{
-				"name": "foo",
+				"name": name,
 			},
 		},
 		Status: api.NodeStatus{
 			Capacity: api.ResourceList{
-				api.ResourceName(api.ResourceCPU):    resource.MustParse("10"),
-				api.ResourceName(api.ResourceMemory): resource.MustParse("0"),
+				api.ResourceCPU:    resource.MustParse("10"),
+				api.ResourceMemory: resource.MustParse("0"),
```

Comment:
Ideally this should return a functional object which can pass validation and be used in code.  If you apply this cleanup to other test-cases you will start to find all those fields.  Not blocking this PR for it, just direction.

Reply:
Not sure how to exactly address this, but it's something I can improve on future PRs once I figure out the specifics.